Libraries/pbxbuild/Sources/Tool/Invocation.cpp

```diff
@@ -115,7 +115,7 @@ Absolute(std::string const &path)
 Executable Executable::
 Builtin(std::string const &name)
 {
-    std::string executableRoot = FSUtil::GetDirectoryName(ProcessContext::GetDefault()->executablePath());
+    std::string executableRoot = FSUtil::GetDirectoryName(ProcessContext::GetDefaultUNSAFE()->executablePath());
```

Comment:
This code is called deep inside pbxbuild, which would be hard to pass down. (Same for the filesystem it uses elsewhere.) Instead, I think this needs to essentially record which constructor was called, then the executor would use the process context and filesystem to actually find the executable.